rcljava/src/main/cpp/org_ros2_rcljava_RCLJava.cpp

```diff
@@ -152,9 +153,30 @@ Java_org_ros2_rcljava_RCLJava_nativeGetRMWIdentifier(JNIEnv * env, jclass)
   return env->NewStringUTF(rmw_implementation_identifier);
 }
 
+#define RCLJAVA_QOS_SET_RMW_TIME(qos_profile, policy_name, seconds, nanos) \
+  do { \
+    if (seconds < 0) { \
+      rcljava_throw_exception( \
+        env, "java/lang/IllegalArgumentException", "negative " #policy_name); \
+      free(qos_profile); \
+      return 0; \
+    } \
+    if (nanos < 0) { \
+      rcljava_throw_exception( \
+        env, "java/lang/IllegalStateException", \
```

Comment:
I think the type of the exception should be the same as the one above: `java/lang/IllegalArgumentException`.

Reply:
The difference is that nanoseconds must always be positive (https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#getNano--), except something is corrupted.
In the case of `seconds`, illegal arguement is right, as negative durations are possible but not accepted in QoSProfile.

I don't mind changing the exception if you think it's better.

Reply:
[`IllegalStateException`](https://docs.oracle.com/javase/8/docs/api/java/lang/IllegalStateException.html) still seems like a strange choice as it is described:

> Signals that a method has been invoked at an illegal or inappropriate time.

Even though nanoseconds internal to `Duration` must always be positive, you can technically [pass a negative value for nanoseconds](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html#ofNanos-long-). Regardless, in the rcljava API proposed here, we're enforcing that the value must be positive, so I think an `IllegalArgumentException` is an appropriate choice.

Reply:
> Even though nanoseconds internal to Duration must always be positive, you can technically pass a negative value for nanoseconds

Yes, and when that happens, `getNano` has to still return a positive number.

Reply:
See https://github.com/osrf/ros2_java/pull/20/commits/b19ac0bbd8e1eaf6502315034de79603abaa53b6.

Reply:
I think it's best to not assume that the nanoseconds passed to this API came from `Duration.getNano()`.

Reply:
Ah, I see your point.
I've made the error messages symmetric in https://github.com/osrf/ros2_java/pull/20/commits/bc788101b71b17f5ee53c87776d1bf7cb3dfa3e4.